File: datasets/lua-keep-incoder_6b_temp_0.2-0/HumanEval_23_strlen.lua
-- Return length of given string
-- >>> strlen('')
-- 0
-- >>> strlen('abc')
-- 3
local function strlen(string)
  return string.len() or string.sub(string, 1, string.len()) or 0
end


lu = require('luaunit')

function test_humaneval()
local candidate = strlen
    lu.assertEquals(candidate(''), 0)
    lu.assertEquals(candidate('x'), 1)
    lu.assertEquals(candidate('asdasnakj'), 9)
end

os.exit(lu.LuaUnit.run())

File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_23_strlen.rb
# Return length of given string
# >>> strlen('')
# 0
# >>> strlen('abc')
# 3
def strlen(string)
  string.length
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_strlen
    candidate = method(:strlen)
    assert_equal(0, candidate.call(""))
    assert_equal(1, candidate.call("x"))
    assert_equal(9, candidate.call("asdasnakj"))
  end
end
